Update Globalnet doc to include hostnetwork support (#99)
submariner-io/submariner-website: src/content/architecture/globalnet/_index.en.md

To support overlapping CIDRs in clusters connected through submariner, submariner has a component called Global Private Network, GlobalNet (`globalnet`). This GlobalNet is a virtual network specifically to support submariner's multi-cluster solution with a Global CIDR. Each cluster is given a subnet from this Global Private Network, configured as new cluster parameter `GlobalCIDR` (e.g. 169.254.0.0/16) which is configurable at time of deployment

> Once configured, each Service and Pod that requires cross-cluster access is allocated an IP, named `globalIp`, from this `GlobalCIDR` that is annotated on the Pod/Service object. This globalIp is used for all cross-cluster communication to and from a Pod and the globalIp of a remote Service. Routing and IPTable rules are configured to use the globalIp for ingress and egress. All address translations occur on the Gateway node.

Reviewer: I was wondering if `globalIp` should be capitalized too 😄 My conclusion was not to as `Pod` is an entity/object while `globalIp` is an attribute/property. But II don't know - we can  capitalized `globalIp` as well - just as long as we're connsistent.

> Once configured, each service and pod that requires cross-cluster access is allocated an IP, named `globalIp`, from this `GlobalCIDR` that is annotated on the Pod/Service object. This `globalIp` is used for all cross-cluster communication to and from to the Pod to globalIp of remote Service. Routing and IPTable/OVS/OVN rules are configured to use this IP for ingress and egress. All address translations occur on the Gateway node.

Reviewer: ```suggestion
Once configured, each Service and Pod that requires cross-cluster access is allocated an IP, named `globalIp`, from this `GlobalCIDR` that is annotated on the Pod/Service object. This globalIp is used for all cross-cluster communication to and from a Pod and the globalIp of a remote Service. Routing and IPTable/OVS/OVN rules are configured to use the globalIp for ingress and egress. All address translations occur on the Gateway node.
```

I think we need to be consistent with our use of case with pods/services. There's a mix of _pod_, _Pod_, and _POD_. We need to pick one. Looking at the k8s docs, they consistently use _Pod_ and _Service_.

Author: Yeah makes sense @tpantelis.

![Figure 1 - Proposed solution](/images/globalnet/overlappingcidr-solution.png)

{{% notice info %}}

> Unlike vanilla Submariner, where POD to POD connectivity is also supported, Globalnet only supports POD to remote service connectivity using globalIps.

Reviewer: ```suggestion
Unlike vanilla Submariner, where Pod to Pod connectivity is also supported, Globalnet only supports Pod to remote Service connectivity using globalIps.
```

Author: Done

{{% /notice %}}

### submariner-globalnet

> Submariner GlobalNet is a component that provides cross-cluster connectivity from PODs to remote Services using their globalIps. Compiled as binary `submariner-globalnet`, it is responsible for maintaining a pool of global IPs, allocating IPs from the GlobalIp pool to pods and services, annotating services and pods with their globalIp, and configuring the required rules on the gateway node to provide cross-cluster connectivity using globalIps.

Reviewer: ```suggestion
Submariner GlobalNet is a component that provides cross-cluster connectivity from Pods to remote Services using their globalIps. Compiled as binary `submariner-globalnet`, it is responsible for maintaining a pool of global IPs, allocating IPs from the GlobalIp pool to pods and services, annotating Services and Pods with their globalIp, and configuring the required rules on the gateway node to provide cross-cluster connectivity using globalIps.
```

Author: Done

> Globalnet also supports connectivity from the Nodes (including PODs that use HostNetworking) to globalIp of remoteServices.

Reviewer: ```suggestion
Globalnet also supports connectivity from the Nodes (including Pods that use HostNetworking) to globalIp of remote Services.
```

Author: Done

It mainly consists of two key components: the IP Address Manager and Globalnet.

#### IP Address Manager (IPAM)

The IP Address Manager (IPAM) component does the following:

* Creates a pool of IP addresses based on the `GlobalCIDR` configured on cluster.
* On creation of a Pod/Service, allocates a globalIp from the GlobalIp pool.
* Annotates the Pod/Service with `submariner.io/globalIp=<global-ip>`.
* On deletion of a Pod/Service, releases its globalIp back to the pool.

#### Globalnet

This component is responsible for programming the routing entries, IPTable rules and does the following:

* Creates initial IPTables chains for Globalnet rules.

> * Whenever a pod is annotated with Global IP, creates an egress SNAT rule to convert source ip from PodIp to pod's globalIp on the Gateway Node.

Reviewer: ```suggestion
* Whenever a Pod is annotated with a globalIp, creates an egress SNAT rule to convert the source Ip from the Pod's Ip to the Pod's globalIp on the Gateway Node.
```

Author: Done

> * Whenever a service is annotated with Global IP, creates an ingress rule to direct all traffic destined to service's globalIp to the service's `kube-proxy` IPTables chain which in turn directs traffic to service's backend pods.

Reviewer: ```suggestion
* Whenever a Service is annotated with a globalIp, creates an ingress rule to direct all traffic destined to the Service's globalIp to the Service's `kube-proxy` IPTables chain which in turn directs traffic to Service's backend Pods.
```

Author: Done

Connectivity is only part of the solution as pods still need to know the IPs of services on remote clusters.

> This is achieved by enhancing [lighthouse](https://github.com/submariner-io/lighthouse) with support for GlobalNet. The Lighthouse controller adds the service's globalIp to the `MultiClusterService` object that is distributed to all clusters. The [lighthouse plugin](https://github.com/submariner-io/lighthouse/tree/master/plugin/lighthouse) then uses the service's globalIp when replying to DNS queries for the service.

Reviewer: ```suggestion
This is achieved by enhancing [lighthouse](https://github.com/submariner-io/lighthouse) with support for GlobalNet. The Lighthouse controller adds the service's globalIp to the `MultiClusterService` object that is distributed to all clusters. The [lighthouse plugin](https://github.com/submariner-io/lighthouse/tree/master/plugin/lighthouse) then uses the Service's globalIp when replying to DNS queries for the Service.
```

Author: Done
